lib/src/license_detection/normalizer.dart

```diff
@@ -0,0 +1,125 @@
+/// Normalizes text according to [SPDX guidelines v2.1][]
+///
+/// [SPDX guidelines v2.1] : https://github.com/spdx/spdx-spec/blo b/v2.2/chapters/appendix-II-license-matching-guidelines-and-templates.md
+String normalize(String text) {
+  // Guideline 4.1.1: upper case and lower case letters should be treated as lower case letters
+  text = text.toLowerCase();
+
+  text = text.replaceAll(_newLineRegex, '\n');
```

Comment:
If you replace `\r` with a new line, then windows line endings will cause doubling of the number of lines,..